### PR TITLE
[IMPROVEMENT] Migrate spotlight call to REST API

### LINKED ISSUE #1449: [IMPROVEMENT] Move "Spotlight" call from RealTime to REST
This API was introduced in 0.61.
Documentation: https://rocket.chat/docs/developer-guides/rest-api/spotlight/.

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Closes #1449

### CODE REVIEW

File: Rocket.ChatTests/API/Requests/General/SpotlightRequestSpec.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        XCTAssertEqual(request.url, urlComponents?.url, "url is correct")
+        XCTAssertEqual(request.httpMethod, "GET", "http method is correct")
+    }
+
+    func testProperties() {
+        let jsonString = """
+        {
+          "users": [
+            {
```
[the remark this answers is not in the document]
Comment (cardoso) [reply Rocket.ChatTests/API/Requests/General/SpotlightRequestSpec.swift]: 👎

File: Rocket.ChatTests/API/Requests/General/SpotlightRequestSpec.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
+        urlComponents?.query = "query=test"
+
+        XCTAssertEqual(request.url, urlComponents?.url, "url is correct")
+        XCTAssertEqual(request.httpMethod, "GET", "http method is correct")
+    }
+
+    func testProperties() {
+        let jsonString = """
+        {
```
[the remark this answers is not in the document]
Comment (cardoso) [reply Rocket.ChatTests/API/Requests/General/SpotlightRequestSpec.swift]: :-1:

**Comment by rafaelks** [comment]
@cardoso Conflicts! 🙈

**Comment by rafaelks** [comment]
@cardoso Conflicts, after merging our extension. 🤦‍♂️

**Comment by rafaelks** [comment]
Nice one!

**Pull request merged**